### PR TITLE
2023-07-monthly-report: Add base report

### PR DESCRIPTION
**Pull request by CohenArthur**
@philberty @P-E-P: do you have something you would like to talk about in the Detailed Changelog section?
@tamaroning @mahadmuhammad: ping for review :)

### CODE REVIEW

**Comment by philberty** [comment]
I've raised https://github.com/Rust-GCC/Reporting/pull/101/files#diff-48ef69aa79f54280fd66dd05b2f2a8bceaf0aa0523eed1c7aac2a91c5563646e

But i will add more stuff here directly then

**Comment by philberty** [comment]
I've added a bunch of changelog entries here

**Comment by philberty** [comment]
actually there is one more to add

**Comment by philberty** [comment]
ok now i am done

**Comment by P-E-P** [comment]
> @philberty @P-E-P: do you have something you would like to talk about in the Detailed Changelog section? @tamaroning @MahadMuhammad: ping for review :)

I would love to write some detailed changelog about proc macros but the changes this month are not very interesting, most of it was additional callback registration as well as new error detection. Nothing too fancy. The previous month would have fitted better but I forgot. I will probably write one later once procedural macros can be resolved properly.

**Comment by tamaroning** [comment]
> @tamaroning @MahadMuhammad: ping for review :)

Looks great now. I love it!

**Comment by MahadMuhammad** [comment]
> Changelog section? @tamaroning @MahadMuhammad: ping for review :)

That's great, LGTM :)

File: 2023-07-monthly-report.org [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+2. E0571
+3. E0268
+4. E0045
+5. E0323
+6. E0423
+7. E0133
+8. E0658
+9. E0093
+10. E0124
+11. E0070
```
Comment (MahadMuhammad) [inline 2023-07-monthly-report.org:273]: There are some error codes, which are being mentioned in above PR's but not here:

```rust
12. E0063
13. E0045
14. E0053
15. E0046
16. E0061
17. E0229
18. E0034

**Review by P-E-P — APPROVED** [review]

File: 2023-07-monthly-report.org (hunk 1 above)
Comment (philberty) [reply 2023-07-monthly-report.org:273]: Feel free to add them @MahadMuhammad

Comment (MahadMuhammad) [reply 2023-07-monthly-report.org:273]: I'll raise the PR when this PR(https://github.com/Rust-GCC/Reporting/pull/102) gets merged or should I raise the PR with all the changes (including new changes) in this file?

Comment (CohenArthur) [reply 2023-07-monthly-report.org:273]: I added them @MahadMuhammad

**Pull request merged**